### PR TITLE
DPE-75 Add lint and build workflows

### PR DESCRIPTION
**Pull request by marceloneppel**
This PR adds lint and build workflows to validate the Dockerfile from this repository.

One addition workflow job related to testing the deploy of the built docker image will be created in another PR.

There are some changes in the Dockerfile that the lint needs.

### CODE REVIEW

**Review by paulomach — APPROVED** [review]
Nice and short! Juan will be happy.
LGTM

**Review by zmraul — APPROVED** [review]
LGTM!

File: .github/workflows/pull-request.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+      - name: Build Image
+        uses: docker/build-push-action@v2
+        with:
+          context: ./
+          file: ./Dockerfile
+          builder: ${{ steps.buildx.outputs.name }}
+          # Do not publish the image.
+          push: false
+          # Set the tag.
```
Comment (MiaAltieri) [inline .github/workflows/pull-request.yaml]: Comment could be more descriptive. I'm new to this so maybe its obvious, but why is setting the tag important?

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+name: Lint (Hadolint)
```
Comment (MiaAltieri) [inline .github/workflows/lint.yaml:5]: A comment before this file would be helpful, perhaps something like:
`# Linter for CI for patroni docker image`

Also not sure if its required but you might want the:
```
# Copyright 2021 Canonical Ltd.
# See LICENSE file for licensing details.
```

File: .github/workflows/pull-request.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+name: Build
```
Comment (MiaAltieri) [inline .github/workflows/pull-request.yaml:5]: A comment before this file would be helpful, perhaps something like:
`# CI testing for building patroni docker image`

Also not sure if its required but you might want the:
```
# Copyright 2021 Canonical Ltd.
# See LICENSE file for licensing details.
```

**Review by MiaAltieri — COMMENTED** [review]
Looks great! Might want to add some comments for certain steps to make things a little more clear, but also I don't know if thats really necessary, I trust your judgement :)

**Review by jnsgruk — APPROVED** [review]
LGTM apart from a few nits above! :)

File: .github/workflows/lint.yaml (hunk 2 above)
Comment (marceloneppel) [reply .github/workflows/lint.yaml:5]: Great addition, Mia. I believe that it's not required to add the copyright notice, but looks better to have it. I also added the LICENSE file as we are commenting about it in this header.

File: .github/workflows/pull-request.yaml (hunk 3 above)
Comment (marceloneppel) [reply .github/workflows/pull-request.yaml:5]: Thanks Mia. Same as the comment above.

File: .github/workflows/pull-request.yaml (hunk 1 above)
Comment (marceloneppel) [reply .github/workflows/pull-request.yaml]: I agree. The comment isn't so clear. We need the tag to recover the built docker image in later steps. As we don't have the deploy step in this PR (only in #5), I'm removed the tag from here and I'll add a better comment there.

**Pull request merged**